docs/data/base/components/click-away-listener/click-away-listener.md

```diff
@@ -0,0 +1,58 @@
+---
+product: base
+title: Detect click outside React component
+components: ClickAwayListener
+githubLabel: 'component: ClickAwayListener'
+packageName: '@mui/base'
+---
+
+# Click-away listener
+
+<p class="description">Detect if a click event happened outside of an element. It listens for clicks that occur somewhere in the document.</p>
+
+- 📦 [981 B gzipped](/size-snapshot).
+- ⚛️ Supports portals
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+## Example
+
+For instance, if you need to hide a menu dropdown when people click anywhere else on your page:
+
+{{"demo": "ClickAway.js"}}
+
+Notice that the component only accepts one child element.
+You can find a more advanced demo on the [Menu documentation section](/components/menus/#menulist-composition).
+
+## Portal
+
+The following demo uses [`Portal`](/base/react-portal/) to render the dropdown into a new "subtree" outside of current DOM hierarchy.
+
+{{"demo": "PortalClickAway.js"}}
+
+## Leading edge
+
+By default, the component responds to the trailing events (click + touch end).
+However, you can configure it to respond to the leading events (mouse down + touch start).
+
+{{"demo": "LeadingClickAway.js"}}
+
+> ⚠️ In this mode, only interactions on the scrollbar of the document is ignored.
```

Comment:
@oliviertassinari I saw you introduced this comment. Would you mind elaborating on it? It seems that clicking on a scrollbar doesn't trigger onClickAway in any mode (at least in my setup - Edge on Windows)